daly-bms-uart.cpp

```diff
@@ -39,63 +38,451 @@ bool Daly_BMS_UART::Init()
     return true;
 }
 
-bool Daly_BMS_UART::getPackMeasurements(float &voltage, float &current, float &SOC)
+bool Daly_BMS_UART::updateAllValues()
+{
+    // There are better ways to write this, but this is fine...
+    if (!getPackMeasurements())
+    {
+        return false; // 0x90
+    }
+    if (!getMinMaxCellVoltage())
+    {
+        return false; // 0x91
+    }
+    if (!getPackTemp())
+    {
+        return false; // 0x92
+    }
+    if (!getDischargeChargeMosStatus())
+    {
+        return false; // 0x93
+    }
+    if (!getStatusInfo())
+    {
+        return false; // 0x94
+    }
+    if (!getCellVoltages())
+    {
+        return false; // 0x95
+    }
+    if (!getCellTemperature())
+    {
+        return false; // 0x96
+    }
+    if (!getCellBalanceState())
+    {
+        return false; // 0x97
+    }
+    if (!getFailureCodes())
+    {
+        return false; // 0x98
+    }
+
+    return true;
+}
+
+bool Daly_BMS_UART::getPackMeasurements() // 0x90
 {
     this->sendCommand(COMMAND::VOUT_IOUT_SOC);
 
     if (!this->receiveBytes())
     {
 #ifdef DALY_BMS_DEBUG
-        Serial.printf("<DALY-BMS DEBUG> Receive failed, V, I, & SOC values won't be modified!\n");
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, V, I, & SOC values won't be modified!\n");
+#endif
+        return false;
+    }
+    get.packVoltage = ((float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]) / 10.0f);
+    get.packCurrent = ((float)(((this->my_rxBuffer[8] << 8) | this->my_rxBuffer[9]) - 30000) / 10.0f);
+    get.packSOC = ((float)((this->my_rxBuffer[10] << 8) | this->my_rxBuffer[11]) / 10.0f);
+#ifdef DALY_BMS_DEBUG
+
+    DEBUG_SERIAL.println("<DALY-BMS DEBUG> " + (String)get.packVoltage + "V, " + (String)get.packCurrent + "A, " + (String)get.packSOC + "SOC");
+    // debug current anomaly
```

Comment:
following code will be removed when cleared what is with the bug that  randomly came a 2795A value